File: starterkits/python/game_command.py
from dataclasses import dataclass
from dataclasses_json import dataclass_json
from enum import Enum
from game_message import Position, UnitType
from typing import List


@dataclass_json
class ActionType(Enum):
    BUY = "BUY"
    UNIT = "UNIT"


@dataclass_json
class Action:
    type: ActionType

    def __init__(self, type: ActionType):
        self.type = type


@dataclass_json
@dataclass
class BuyAction(Action):
    unitType: UnitType
    type: ActionType

    def __init__(self, unit_type: UnitType):
        super().__init__(ActionType.BUY)
        self.type = ActionType.BUY
        self.unitType = unit_type


@dataclass_json
class UnitActionType(Enum):
    MOVE = "MOVE"
    ATTACK = "ATTACK"
    PICKUP = "PICKUP"
    MINE = "MINE"
    DROP = "DROP"
    NONE = "NONE"


@dataclass_json
@dataclass
class UnitAction(Action):
    target: Position
    action: UnitActionType
    unitId: str
    type: ActionType

    def __init__(self, action: UnitActionType, unit_id: str, target: Position):
        super().__init__(ActionType.UNIT)
        self.action = action
        self.unitId = unit_id
        self.target = target
        self.type = ActionType.UNIT


@dataclass_json
class GameCommand:
    actions: List[Action]

    def __init__(self, actions: List[Action]):
        self.actions = actions

File: starterkits/csharp/game/GameCommand.cs
using System.Collections.Generic;
using static Blitz2021.Map;
using static Blitz2021.Unit;

namespace Blitz2021
{
    public class GameCommand
    {
        public abstract class Action
        {
            public enum ActionType
            {
                BUY, UNIT
            }

            public ActionType type;
        }

        public class BuyAction : Action
        {
            public UnitType unitType;

            public BuyAction(UnitType unitType)
            {
                this.type = ActionType.BUY;
                this.unitType = unitType;
            }
        }

        public class UnitAction : Action
        {
            public enum UnitActionType
            {
                MOVE, ATTACK, PICKUP, MINE, DROP, NONE
            }
            public Position target;
            public UnitActionType action;
            public string unitId;

            public UnitAction(UnitActionType action, string unitId, Position target)
            {
                this.type = ActionType.UNIT;
                this.unitId = unitId;
                this.target = target;
                this.action = action;
            }
        }

        public GameCommand(List<Action> actions)
        {
            this.actions = actions;
        }

        public List<Action> actions;
    }
}